app/src/androidTest/java/com/sdp/movemeet/view/profile/EditProfileActivityDeleteAccountTest.java

```diff
@@ -0,0 +1,117 @@
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static androidx.test.espresso.action.ViewActions.replaceText;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+
+
+@RunWith(AndroidJUnit4.class)
+public class EditProfileActivityDeleteAccountTest {
+
+    private static final String TAG = "DeleteAccountTest";
+
+    public static final String TEST_FULL_NAME = "Yolo Test";
+    public static final String TEST_EMAIL = "yolotest@gmail.com";
+    public static final String TEST_PASSWORD = "123456";
+    public static final String TEST_PHONE = "0798841817";
+
+    private FirebaseAuth fAuth;
+
+    @Before
+    public void signOutAndCreateAccount() {
+
+        // Signing out
+        fAuth = AuthenticationInstanceProvider.getAuthenticationInstance();
+        if (fAuth.getCurrentUser() != null) {
+            fAuth.signOut();
+        }
+
+        try {
+            Thread.sleep(1500);
+        } catch (InterruptedException e) {
+            assert (false);
+        }
+
+        ActivityScenario scenario = ActivityScenario.launch(RegisterActivity.class);
+
+        // Creating an account
+        onView(withId(R.id.edit_text_full_name)).perform(replaceText(TEST_FULL_NAME), closeSoftKeyboard());
+        onView(withId(R.id.edit_text_email)).perform(replaceText(TEST_EMAIL), closeSoftKeyboard());
+        onView(withId(R.id.edit_text_password)).perform(replaceText(TEST_PASSWORD), closeSoftKeyboard());
+        onView(withId(R.id.edit_text_phone)).perform(replaceText(TEST_PHONE), closeSoftKeyboard());
+        onView(withId(R.id.button_register)).perform(click());
+        try {
+            Thread.sleep(1500);
+        } catch (InterruptedException e) {
+            assert (false);
+        }
+
+    }
+
+    @Test
+    public void deleteAccount_2() {
+
+        try (ActivityScenario<EditProfileActivity> scenario = ActivityScenario.launch(EditProfileActivity.class)) {
+
+            scenario.onActivity(activity -> {
+                fAuth = AuthenticationInstanceProvider.getAuthenticationInstance();
+                FirebaseUser firebaseUser = fAuth.getCurrentUser();
+                String userId = firebaseUser.getUid();
+                activity.deleteUserAccount(userId, firebaseUser);
+            });
+
+        } catch (Exception e) {
+            Log.d("TAG", "deleteAccount Exception: " + e);
+            e.printStackTrace();
+        }
+
+        try {
+            Thread.sleep(2000);
+        } catch (InterruptedException e) {
+            assert (false);
+        }
+
+        // Trying to re-login to assert that the user account doesn't exist anymore
+        CountDownLatch latch = new CountDownLatch(1);
+        fAuth = AuthenticationInstanceProvider.getAuthenticationInstance();
+        fAuth.signInWithEmailAndPassword(TEST_EMAIL, TEST_PASSWORD).addOnSuccessListener(new OnSuccessListener<AuthResult>() {
```

Comment:
It could be useful to use this syntax and a generic account for all tests!

Reply:
Completely agree! And refactoring those reference Strings in TEST_EMAIL and TEST_PASSWORD variables makes the test clearer!

Reply:
We can do it next week!